src/ast.js

```diff
@@ -90,6 +117,88 @@ export class AST {
+      let removedCount = 0, fromNode = this.getNodeAfterCur(from), toNode = this.getNodeBeforeCur(to);
+      if(fromNode && toNode) { // we might have removed something. Check to see.
+        // find the nodes touching from & to, and see how many siblings fall between them (+ 1 if same node)
+        removedCount = fromNode.path.split(",").pop() - fromNode.path.split(",").pop() + 1;
+        if(removed == "") removedCount--; // if we're strictly inserting, subtract 1
+      }
+      oldAST.nodeIdMap.forEach(n => {
+        n.from = adjustForChange(n.from, change, true );
+        n.to   = adjustForChange(n.to,   change, false);
+      });
+      return {path: path, added: insertedSiblings, removed: removedCount };
+    });
+    // for each pathChange, nullify removed nodes and adjust the paths of affected nodes
+    pathChanges.forEach(({path, added, removed}) => {
+      let shift = added - removed;
+      // force a re-render on the parent, since parent nodeType could change
+      if(shift == 0) { oldAST.nodeIdMap.delete(oldAST.getNodeByPath(path).id); return; }
+      let cArray = path.split(',').map(Number);
+      let changeDepth = cArray.length-1, changeIdx = cArray[changeDepth];
+      oldAST.nodeIdMap.forEach((node, id) => {
+        let pArray = node.path.split(',').map(Number);
+        // if the node is independent of the change, just return
+        if(pathIsIndependentOfChangePath(pArray, cArray)) { return; }
+        // If it's being removed, delete from nodeIdMap and mark its parent as dirty (if it has one)
+        if(pArray[changeDepth] < (changeIdx + removed)) {
+          let parent  = oldAST.getNodeParent(node);
+          if(parent) { dirtyNodes.add(parent); }
+          oldAST.nodeIdMap.delete(id);
+        // Otherwise just update the path of other post-change nodes by +shift
+        } else {
+          pArray[changeDepth] += shift;
+          node.path = pArray.join(',');
+        }
+      });
+    });
+    // copy over the DOM elt for unchanged nodes, and update their IDs to match
+    oldAST.nodeIdMap.forEach(n => {
+      let newNode = newAST.getNodeByPath(n.path);
+      if(newNode) { n.el.id = 'block-node-' + newNode.id; newNode.el = n.el; }
+    });
+    // If we have a DOM elt, use it and update the id. Mark parents of nodes with DOM elts as dirty
+    newAST.nodeIdMap.forEach(n => { if(!n.el){ dirtyNodes.add(newAST.getNodeParent(n) || n); }});
+    // Ensure that no dirty node is the ancestor of another dirty node
+    let dirty = [...dirtyNodes].sort((a, b) => a.path<b.path? -1 : a.path==b.path? 0 : 1);
+    dirty.reduce((n1, n2) => n2.path.includes(n1.path)? dirtyNodes.delete(n2) && n1 : n2, false);
+    newAST.dirtyNodes = new Set([...dirtyNodes].map(n => newAST.getNodeByPath(n.path)) // grab all the nodes
+      .filter(n => n !== undefined));                                                  // remove deleted ones
+    return newAST;
+  }
+
```

Comment:
Where is all this coming from? Maybe leftover from basing off `new-reactify-with-tests`?  This file shouldn't have any of these changes. It should be reset the to what it is on `new-reactify`.